src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs

```diff
@@ -1162,6 +1163,7 @@ public static MethodCallExpression Call(Expression instance, string methodName,
         /// <exception cref="ArgumentNullException">
         /// <paramref name="type"/> or <paramref name="methodName"/> is null.</exception>
         /// <exception cref="InvalidOperationException">No method whose name is <paramref name="methodName"/>, whose type parameters match <paramref name="typeArguments"/>, and whose parameter types match <paramref name="arguments"/> is found in <paramref name="type"/> or its base types.-or-More than one method whose name is <paramref name="methodName"/>, whose type parameters match <paramref name="typeArguments"/>, and whose parameter types match <paramref name="arguments"/> is found in <paramref name="type"/> or its base types.</exception>
+        [RequiresUnreferencedCode("Calling a generic method cannot be statically analyzed. It's not possible to guarantee the availability of requirements of the generic method.")]
         public static MethodCallExpression Call(
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)] Type type,
```

Comment:
I actually don't know what will happen tot he annotation in this case. Do we need the annotation if the method is unsafe?

Reply:
Similar comment below on other methods.

Reply:
Opened https://github.com/mono/linker/issues/1815 for this.

These annotations are valid IMO. The type should get the correct methods preserved on it. But the caller should still get a warning because of the `Type[] typeArguments` may be non-null, in which case the Expressions code will call `MakeGenericMethod`. Callers can suppress this if `null` or an empty array is passed in.

Reply:
> Callers can suppress this if null or an empty array is passed in.

Would it make sense to put that guidance into the RequiresUnreferencedCode message?

Reply:
Done.